### ISSUE TITLE
stride-api: Add an api that will add the "src_city" and "dst_city" for every gtfs_route

### ISSUE THREAD

**Issue by ShayAdler** [hasadna/open-bus-map-search#311]
Currently, we only have the "route_long_name", which is a string containg the first and last stops nickname, and a seperating string -  `<->`. For example, this is how we currently use it in the UI - 
<img width="555" alt="image" src="https://github.com/hasadna/open-bus-map-search/assets/61648359/159b55e6-538a-46d6-98bb-8f094888773a">

In some routes, the city name appears in the string, but as far as I know it is not guranteed. 
So, 2 suggestions I have here, but fill free to suggest more:
* Get the coordinates of the first and last stop in each route (using [gtfs_ride_stop.stop_sequence](https://github.com/hasadna/open-bus-stride-db/blob/main/DATA_MODEL.md#gtfs_ride_stopstop_sequence)), and using google [geo api](https://developers.google.com/maps/documentation/geocoding/overview) (or other options), query the city name.
* Using GPT, when the input is the split route_long_name. I think this is a bit of a long shot that will create some false positives when the exact city name is not in the route_long_name, but mentioning this anyway as an option 😅 

An even greater implementation could be adding an ETL, that will add a column in the DB to the gtfs_route table, so that this logic won't be done in runtime, and reachable by any of the APIs.

**Comment by ShayAdler** [comment]
@NoamGaash @ArkadiK94 lmk if you think of any smaller separation that can be done here :)

**Comment by ShayAdler** [comment]
@NoamGaash suggested that instead of using geo api (which requires and api key), you can download a mapping of the cities in Israel to their poligon, store it hard coded, and search for the nearest neighbour. 
https://simplemaps.com/data/il-cities

**Comment by oribro** [comment]
@NoamGaash I'd like to tackle this issue

**Comment by ShayAdler** [comment]
@oribro assigning you :)
Lmk if you need any help

**Comment by oribro** [comment]
@ShayAdler PM'd you on Slack

**Comment by NoamGaash** [comment]
@oribro right, after changing the backend endpoint, we'll need to fix the FE type definitions as well

**Comment by shootermv** [comment]
did the backend already changed?

**Comment by NoamGaash** [comment]
no. that's the task.

**Comment by ShayAdler** [comment]
@oribro - Regarding GPT approach - in my opinion should only use that as a last resort, cause it has many pitfalls (mainly latency and inconsistency).

**Comment by NoamGaash** [comment]
@oribro let us know where are you standing and if you need any help! If you're still on this issue, please let us know and you'll be re-assigned
Thanks!

**Comment by oribro** [comment]
@NoamGaash This is the issue I mentioned. Could use some help

**Comment by NoamGaash** [comment]
Sure, @oribro, do you have any specific question that I can try to answer on?
